api/app/weather_models/env_canada.py

```diff
@@ -458,6 +467,27 @@ def _process_prediction(self,  # pylint: disable=too-many-arguments
             points, prediction.tmp_tgl_2, coordinate, method='linear')[0]
         station_prediction.rh_tgl_2 = griddata(
             points, prediction.rh_tgl_2, coordinate, method='linear')[0]
+        # Check that apcp_sfc_0 is not None, since accumulated precipitation
+        # does not exist for 00 hour
+        if prediction.apcp_sfc_0 is not None:
+            station_prediction.apcp_sfc_0 = griddata(
+                points, prediction.apcp_sfc_0, coordinate, method='linear')[0]
+        else:
+            station_prediction.apcp_sfc_0 = 0.0
+        # Calculate the delta_precipitation based on station's previous prediction_timestamp
+        # for the same model run
+        # For some reason pylint doesn't think session has a flush!
+        self.session.flush()  # pylint: disable=no-member
+        results = self.session.query(WeatherStationModelPrediction).\
+            filter(WeatherStationModelPrediction.station_code == station.code).\
+            filter(WeatherStationModelPrediction.prediction_model_run_timestamp_id == model_run.id).\
+            filter(WeatherStationModelPrediction.prediction_timestamp < prediction.prediction_timestamp).\
+            order_by(WeatherStationModelPrediction.prediction_timestamp.desc()).\
+            first()
+        if results is not None:
+            station_prediction.delta_precip = station_prediction.apcp_sfc_0 - results.apcp_sfc_0
```

Comment:
Our assumption on when precipitation resets may be incorrect. See below - the total precipitation resets to 0 at 20h00 UTC. (Suspicious, that's 12pm PDT...)

{
              "datetime": "2020-12-08T18:00:00+00:00",
              "temperature": 6.673656113942512,
              "bias_adjusted_temperature": 4.908975877187613,
              "dew_point": null,
              "relative_humidity": 94.2193923950193,
              "bias_adjusted_relative_humidity": 87.5482053821786,
              "wind_speed": null,
              "wind_direction": null,
              "total_precipitation": 57.82819366455246,
              "delta_precipitation": 0.481288592020654,
              "accumulated_rain": null,
              "accumulated_snow": null,
              "accumulated_freezing_rain": null,
              "accumulated_ice_pellets": null,
              "cloud_cover": null,
              "sea_level_pressure": null,
              "wind_speed_40m": null,
              "wind_direction_40m": null,
              "wind_direction_80m": null,
              "wind_speed_120m": null,
              "wind_direction_120m": null,
              "wind_speed_925mb": null,
              "wind_direction_925mb": null,
              "wind_speed_850mb": null,
              "wind_direction_850mb": null
            },
            {
              "datetime": "2020-12-08T20:00:00+00:00",
              "temperature": 6.736517238616976,
              "bias_adjusted_temperature": 4.357358128062524,
              "dew_point": null,
              "relative_humidity": 91.97403988308346,
              "bias_adjusted_relative_humidity": 89.93398344290621,
              "wind_speed": null,
              "wind_direction": null,
              "total_precipitation": 0,
              "delta_precipitation": -57.82819366455246,
              "accumulated_rain": null,
              "accumulated_snow": null,
              "accumulated_freezing_rain": null,
              "accumulated_ice_pellets": null,
              "cloud_cover": null,
              "sea_level_pressure": null,
              "wind_speed_40m": null,
              "wind_direction_40m": null,
              "wind_direction_80m": null,
              "wind_speed_120m": null,
              "wind_direction_120m": null,
              "wind_speed_925mb": null,
              "wind_direction_925mb": null,
              "wind_speed_850mb": null,
              "wind_direction_850mb": null
            },

Reply:
"total_precip" at 20:00 UTC is 0 because that's from GDPS, where we're inserting an extra prediction at noon PDT. It's not actually resetting, you'll see the values for 21:00 UTC carry on where 18:00 left off

Reply:
Oh yeah - duh - good point. So we need to interpolate the values here.